src/providers/originProvider.ts

```diff
@@ -0,0 +1,105 @@
+import { CACHE_HEADERS } from '../constants/cache';
+import { Context } from '../context';
+import {
+  GetFileOptions,
+  GetFileResult,
+  HeadFileResult,
+  HttpResponseHeaders,
+  Provider,
+  ReadDirectoryResult,
+} from './provider';
+
+type OriginProviderCtorOptions = {
+  ctx: Context;
+};
+
+/**
+ * Serves assets from origin.nodejs.org, used as a fallback for if R2 fails.
+ */
+export class OriginProvider implements Provider {
+  private ctx: Context;
+
+  constructor({ ctx }: OriginProviderCtorOptions) {
+    this.ctx = ctx;
+  }
+
+  async headFile(path: string): Promise<HeadFileResult | undefined> {
+    const res = await fetch(this.ctx.env.ORIGIN_HOST + path, {
+      method: 'HEAD',
+      headers: {
+        'user-agent': 'release-cloudflare-worker',
```

Comment:
Is the user agent necessary? 🤔 for what do we use it?

Reply:
Not necessary but thought it would be nice to include just in case we need it one day, can remove if it'd be preferable